cli/flox/src/commands/activate.rs

```diff
@@ -301,12 +308,7 @@ impl Activate {
             // Export FLOX_ENV_DESCRIPTION for this environment and let the
             // activation script take care of tracking active environments
             // and invoking the appropriate script to set the prompt.
-            (
-                FLOX_ENV_DESCRIPTION_VAR,
-                now_active
-                    .bare_description()
-                    .expect("`bare_description` is infallible"),
-            ),
+            (FLOX_ENV_DESCRIPTION_VAR, now_active.message_description()?),
```

Comment:
> currently we're passing `bare_description()` for `FLOX_ENV_DESCRIPTION`, do we want to preserve that? I didn't cause this is a message not a prompt

(threading my own response here because this is the related change)

I think it still makes sense to use `bare` and add the quoting in `activate` (or a child script thereof) because having single quotes around the environment variable value feels odd, it restricts how else we or the user we might use it, and will render incorrectly for existing environments until we do #1974.

As an aside which could be followups:

1. It looks like `bare_description` and `message_description` have been renamed and refactored several times but I found it hard to reason about the current split. Why does one report "current directory" and the other doesn't? Could `bare_description` just be called `description` and `message_description` called `quoted_description`, with the former wrapping the latter to add the single quotes?
1. `FLOX_ENV_DESCRIPTION` is also referenced in [start.bash](https://github.com/flox/flox/blob/99d910a4bb924e691f00bcee7b30e9574bc2c5ed/assets/activation-scripts/activate.d/start.bash#L35C1-L35C34) but `FLOX_PROMPT_ENVIRONMENTS_activate` doesn't appear to be used anywhere.

Reply:
All of the `*_activate` variables were temporary, used to verify functionality with a previous refactor. They can all be removed on that basis.

Reply:
I dropped the 2nd commit and added some TODOs about `bare_description` and `message_description`